circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit_init.cpp

```diff
@@ -0,0 +1,195 @@
+#include "aztec3/circuits/kernel/private/init.hpp"
+#include "aztec3/constants.hpp"
+#include "aztec3/utils/array.hpp"
+
+using aztec3::circuits::abis::CombinedConstantData;
+using aztec3::circuits::abis::CombinedHistoricTreeRoots;
+using aztec3::circuits::abis::PrivateHistoricTreeRoots;
+using aztec3::circuits::abis::private_kernel::PrivateKernelInputsInit;
+using aztec3::utils::array_push;
+using aztec3::utils::is_array_empty;
+using aztec3::utils::push_array_to_array;
+using CircuitErrorCode = aztec3::utils::CircuitErrorCode;
+
+namespace aztec3::circuits::kernel::private_kernel {
+
+// using plonk::stdlib::merkle_tree::
+
+// // TODO: NEED TO RECONCILE THE `proof`'s public inputs (which are uint8's) with the
+// // private_call.call_stack_item.public_inputs!
+// CT::AggregationObject verify_proofs(Composer& composer,
+//                                     PrivateKernelInputsInit<CT> const& private_inputs,
+//                                     size_t const& num_private_call_public_inputs,
+//                                     size_t const& num_private_kernel_public_inputs)
+// {
+//     CT::AggregationObject aggregation_object = Aggregator::aggregate(
+//         &composer, private_inputs.private_call.vk, private_inputs.private_call.proof,
+//         num_private_call_public_inputs);
+
+//     Aggregator::aggregate(&composer,
+//                           private_inputs.previous_kernel.vk,
+//                           private_inputs.previous_kernel.proof,
+//                           num_private_kernel_public_inputs,
+//                           aggregation_object);
+
+//     return aggregation_object;
+// }
+
+void initialise_end_values(PrivateKernelInputsInit<NT> const& private_inputs,
+                           KernelCircuitPublicInputs<NT>& public_inputs)
+{
+    // Define the constants data.
+    auto const& private_call_public_inputs = private_inputs.private_call.call_stack_item.public_inputs;
+    auto const constants = CombinedConstantData<NT>{
+        .historic_tree_roots =
+            CombinedHistoricTreeRoots<NT>{
+                .private_historic_tree_roots =
+                    PrivateHistoricTreeRoots<NT>{
+                        .private_data_tree_root = private_call_public_inputs.historic_private_data_tree_root,
+                        .nullifier_tree_root = private_call_public_inputs.historic_nullifier_tree_root,
+                        .contract_tree_root = private_call_public_inputs.historic_contract_tree_root,
+                        .l1_to_l2_messages_tree_root = private_call_public_inputs.historic_l1_to_l2_messages_tree_root,
+                    },
+            },
+        .tx_context = private_inputs.signed_tx_request.tx_request.tx_context,
+    };
+
+    // Set the constants in public_inputs.
+    public_inputs.constants = constants;
+
+    push_array_to_array(private_call_public_inputs.private_call_stack, public_inputs.end.private_call_stack);
```

Comment:
There are other fields in `PrivateCircuitPublicInputs` that might need to be initialised here.